### PR TITLE
fix: Subsetter.parse_resource generator now using is_opendap_url

### LINKED ISSUE #304: Bug with subset on opendap resources
## Description

`Subsetter.parse_resources` has a buggy generator. It returns both the url, and then the path to the local file.

### PR DESCRIPTION
**Pull request by huard**
## Overview

This PR fixes #304

Changes:

* Added `is_opendap_url` function from Finch
* Fixed generator logic
* make_geoms now returns a dict keyed by feature id
* added test for subset_wfs_polygon

## Additional Information

Links to other issues or sources.

### CODE REVIEW

**Comment by cehbrecht** [comment]
@huard looks good to me. Feel free to merge.

I didn't know about the f-strings in python 3.x
https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python

Also there is a new exception pattern in python 3.x:
```
raise Exception(msg) from other_error
```

codacy is not aware hat we use python 3.x only code. 

You could check the other codacy complains ... `import netcdf4`.

**Comment by huard** [comment]
f-strings really make string formatting cleaner.

I've removed 3.5 in the supported versions (xarray dropped support for it last month).

File: tests/test_subset_wfs_polygon.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from flyingpigeon.processes.wps_subset_wfs_polygon import SubsetWFSPolygonProcess
+
+
+def test_wps_subset():
+    client = client_for(Service(processes=[SubsetWFSPolygonProcess()]))
+
+    datainputs = "resource=@xlink:href={nc};" \
+                 "typename={typename};" \
+                 "geoserver={geoserver};" \
+                 "featureids={fid1};" \
```
Comment (Zeitsperre) [inline tests/test_subset_wfs_polygon.py]: The WPS requests expects individual entries per featureid?

Comment (huard) [reply tests/test_subset_wfs_polygon.py]: Not sure I understand the question. You may send multiple featureids. Birdy converts a list or tuple into those individual entries in the HTML request.

Comment (huard) [reply tests/test_subset_wfs_polygon.py]: Any idea how to tell codacy this is python 3 only ?

Comment (cehbrecht) [reply tests/test_subset_wfs_polygon.py]: I've configured the *codacy patterns* to use `PyLint (Python 3)`  ...  but it didn't help.

**Pull request merged**